src/main/java/seedu/address/model/person/Major.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a Person's major in the address book.
+ */
+public class Major {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Major is invalid. Please choose one of the following: ‘cs’, ‘bza’, ‘ceg’, ‘isys’, ‘isec’.";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[a-zA-Z]+";
+
+    public static final List<String> VALID_MAJORS = Arrays.asList("cs", "bza", "ceg", "isys", "isec");
+    private String major;
+
+    /**
+     * Constructs a {@code Major}.
+     *
+     * @param major a valid major.
+     */
+    public Major(String major) {
+        requireNonNull(major);
+        checkArgument(isValidMajor(major), MESSAGE_CONSTRAINTS);
+        this.major = major;
+    }
+
+    /**
+     * Returns if a given major string is a valid role.
+     */
+    public static boolean isValidMajor(String test) {
+        return test.matches(VALIDATION_REGEX) && VALID_MAJORS.contains(test);
+    }
```

Comment:
I'm not certain how regex works, whether its case insensitive or not, but we should make the majors case insensitive since it's what we wrote in the feature list, and I think it's reasonable to have it be case insensitive.

Reply:
Other than the validation being case insensitive, we should also discuss if we want to display is as Cs, CS, or cs on the app itself.